dali/operators/reader/loader/video/frames_decoder.cc

```diff
@@ -206,8 +206,9 @@ FramesDecoder::FramesDecoder(const std::string &filename)
 
 
 FramesDecoder::FramesDecoder(const char *memory_file, int memory_file_size, bool build_index,
-                             bool init_codecs, int num_frames)
+                             bool init_codecs, int num_frames, const std::string &source_info)
```

Comment:
Since we use it to construct a string member, we could pass `source_info` by value and move it to `filename_`. This would save one copy if the constructor call does not use a reference to an existing `string` object. Alternatively, you can use `std::string_view`, which will not create a copy of the argument nor will it construct a temporary string.

Reply:
> we could pass source_info

Done

> move it to filename_

Do we have to move explicitly or the compiler will silently do so?

Reply:
Reworked